### DISCUSSION TITLE
How to get ROAS for a particular channel at different time

### DISCUSSION THREAD

**Discussion by yongqiangzhangzazzle** [google/meridian#229] [Q&A] [answered]
I have trained a MMM model and I can see the ROAS at the model level. However, we'd like to understand how ROAS changes along the time. One thought is to use response curve - if we can get the function of the response curve, then we can apply the function to the spend (say weekly spend) and the get the response and roas. However, I don't find a way to get the function for response curve yet. Any pointers or other suggestions?

**Comment by yongqiangzhangzazzle** [comment]
Just an update that I found two ways to get the response curve data:

1. After training the model, we get:
   media_effects = visualizer.MediaEffects(meridian)
   response_curves_data = media_effects.response_curves_data()

2. After running the optimization:
  df = optimization_result._get_response_curves_data(10)

#1 gives a larger range in spend while #2 gives larger roas values. Wondering which one should I use in this case?

**Comment by brendalewisprice** [comment, marked as answer]
1) Another option for exploring ROAS over time is to take the estimated incremental impact as found in "Analyzer().incremental_impact()" and use the "selected_times" option to choose the week(s) of interest. Dividing by spend over those week(s) will give you the ROAS and will reflect the individual time period more accurately.

2) However, it is important to keep in mind when tracking ROAS over time that even though the coefficients in the model are not time-varying, the ROAS can still change over time because it is dependent on additional factors that might vary across time. For example, the Hill curves model the non-linear, diminishing returns of media execution, therefore, the amount of media execution at a given time can impact the ROAS. Another factor is that media allocation can vary across geos over time with different effectiveness. One more example is that the cost of media execution may vary across time.

3) Using the response curve to calculate ROAS over time using weekly spend is not recommended because the response curve is constructed based on the geo & time flighting pattern for the entire time range and won't be specific to the week you are estimating.

4) The two approaches for outputting the response curves should not return different ROAS estimates. They should be the same unless a different time range was used to estimate the curves, so I'm not sure why you are seeing the discrepancies in ROAS values. 

5) As for the larger spend range for (1) above, that is likely due to the settings in the optimizer. The response curves in the optimizer will be limited by spend range within the spend constraint (i.e. +/- 30% for default).
